### ISSUE TITLE
Python: Add logging statements

### ISSUE THREAD

**Issue by helderco** [dagger/dagger#3806]
There should be more logging statements in key places.

```[tasklist]
### Tasks
- [ ] Info and debug logs during provisioning
- [ ] Example for configuring logging as a user
- [ ] Example for using logs to help debug (e.g., Sentry)
```
```[tasklist]
### Needs exploring
- [ ] Is the debugging from the httpx transport is enough?
- [ ] Does the httpx transport log the responses?
- [ ] Debug log anything in the query builder?
- [ ] Log captured lower level exceptions or is it the user's responsibility to log dagger exceptions?
- [ ] Helper to help configure logging as a user?
```

**Comment by helderco** [comment]
I went slightly out of scope on this one by trying to present logging information in key places to the screen, so the user is able to see a status update on the progress of the pipeline. I showed this in the November 17th [community call](https://dagger-io.zoom.us/rec/play/Yn3iYIh-cru3FAlfHLv68yjbtVrhpVWt0m_uXD-SlWQxUDmIcf6KvSaFReZv-BR343Smb__udXssU_oH.UQL_XIP9QNCNp4CR?continueMode=true&_x_zm_rtaid=E_GlwLxFTrSbGCYgcRCiIw.1670846490283.004f4b991d4ce0f05cdf05b1fd9d5261&_x_zm_rhtaid=724) (0:40:45), almost a month ago.

I'll create a new issue for that specifically and just add the logging statements here to unstuck.
